pandas/core/groupby/groupby.py

```diff
@@ -617,44 +618,52 @@ def _make_wrapper(self, name):
         def wrapper(*args, **kwargs):
             # a little trickery for aggregation functions that need an axis
             # argument
-            kwargs_with_axis = kwargs.copy()
-            if "axis" not in kwargs_with_axis or kwargs_with_axis["axis"] is None:
-                kwargs_with_axis["axis"] = self.axis
-
-            def curried_with_axis(x):
-                return f(x, *args, **kwargs_with_axis)
+            kwargs2 = kwargs.copy()
 
             def curried(x):
-                return f(x, *args, **kwargs)
+                return f(x, *args, **kwargs2)
 
             # preserve the name so we can detect it when calling plot methods,
             # to avoid duplicates
-            curried.__name__ = curried_with_axis.__name__ = name
+            curried.__name__ = name
 
             # special case otherwise extra plots are created when catching the
             # exception below
             if name in base.plotting_methods:
                 return self.apply(curried)
 
-            try:
-                return self.apply(curried_with_axis)
-            except Exception:
-                try:
-                    return self.apply(curried)
-                except Exception:
-
-                    # related to : GH3688
-                    # try item-by-item
-                    # this can be called recursively, so need to raise
-                    # ValueError
-                    # if we don't have this method to indicated to aggregate to
-                    # mark this column as an error
-                    try:
-                        return self._aggregate_item_by_item(name, *args, **kwargs)
-                    except AttributeError:
-                        # e.g. SparseArray has no flags attr
-                        raise ValueError
+            import inspect
```

Comment:
import at the top of the module

Reply:
its an expensive import im trying to get rid of in the one other place we use it

Reply:
really? interesting

Reply:
What do we consider an "expensive" import? It seems natural to me to import the `stdlib` at the top of a module so would prefer to do that as well unless there really is a lot to be gained here

Reply:
About 2.6 ms (of about 400 ms total).  So its not massive, but at this point it is among the lower-hanging fruit.  Fine to move I guess.